### ISSUE TITLE
Add "Marked Style N text to Clipboard" menu commands

### ISSUE THREAD

**Issue by donho** [notepad-plus-plus/notepad-plus-plus#8943]
Add "Marked Style N text to Clipboard" (N:1..5) &  "All Marked text to Clipboard" 6 commands.
So user can mark text with any style and copy them to the clipboard.

**Comment by sasumner** [comment]
Would this be to copy only _unique_ text?

Example:
If a user styles one text `Jack` in a document, ALL occurrences get the styling:
![image](https://user-images.githubusercontent.com/30118311/94950600-e0f77100-04b0-11eb-9a85-4b1aabad626c.png)
Is there much use in copying this data THREE times?

**Comment by donho** [comment]
@sasumner 
Here is an example:
![image](https://user-images.githubusercontent.com/90293/94951456-9394f700-04e4-11eb-85fc-7e08d33d8705.png)

**Comment by sasumner** [comment]
@donho 

Yes, I understand the core concept, but for your example, should output (clipboard contents) for the "all" case be (A):

```
Notepad++
source code editor
GNU General Public License
Notepad++
Notepad++
a greener environment
```

Or (B):

```
Notepad++
source code editor
GNU General Public License
a greener environment
```

I think it should be (B).
User likely won't want redundant data, but has no way to NOT get all the duplicates at time of doing the styling.

**Comment by sasumner** [comment]
For naming suggestion for new commands, see https://github.com/notepad-plus-plus/notepad-plus-plus/pull/8867#discussion_r498952885

**Comment by ArkadiuszMichalski** [comment]
Copy bookmarks allows duplicates ...

**Comment by donho** [comment]
(A) "all" case

It's for user to eliminate the redundant text if they don't want them.

**Comment by donho** [comment]
OTOH, I'm not sure where is suitable to put these commands.
It could be in `Edit >> Copy to Clipboard` or add `Copy to Clipboard` just behind `Search >> Jump down`.

**Comment by sasumner** [comment]
>It's for user to eliminate the redundant text if they don't want them.

OK...
But they will ask: 
"I only selected `Notepad++` once, how come I got 25 copies of it when I pasted?"
"How can I remove the redundant text that happened when I pasted?"
And they will not like the answer when someone starts talking about regular expression replacement. :-)

It could be a problem, because the core of the "styling" feature really is all about multiple occurrences being visually highlighted.

Just pointing out the potential problems.

Possible ideas (for sometime in the future, if problems result):

* Preference setting so that styling only styles the currently selected text (not any of its duplicates)
* Preference setting so that redundant data is not copied

**Comment by sasumner** [comment]
>I'm not sure where is suitable to put these commands.
It could be in Edit >> Copy to Clipboard or add Copy to Clipboard just behind Search >> Jump down (all the Mark operations are in this group).

FWIW, some user input on that:  https://github.com/notepad-plus-plus/notepad-plus-plus/issues/6095#issuecomment-631107161

**Comment by sasumner** [comment]
BTW, I had this idea (the topic of this issue) over a year ago: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/6095#issuecomment-525345377

:-)

**Comment by sasumner** [comment]
>Copy bookmarks allows duplicates ...

Maybe it shouldn't :-)

(No, I'm not advocating changing it)

**Comment by ArkadiuszMichalski** [comment]
Adding next commands without duplicate would be too far...or maybe not:) But it's easier to get rid of them later than to recreate them.

**Comment by sasumner** [comment]
What is thought of this menu text? :

![image](https://user-images.githubusercontent.com/30118311/94969832-67bc4600-04d1-11eb-963b-8ce0a1a60ff6.png)

I dropped the `to Clipboard` from all of them because it felt too "heavy".

The `(Find)` in _(Find) Marked text_ is an effort to stay somewhat common with this menu:

![image](https://user-images.githubusercontent.com/30118311/94970006-b79b0d00-04d1-11eb-8a65-6fb31312d410.png)

...but keeps the emphasis on "Marked"

**Comment by ArkadiuszMichalski** [comment]
This "Find style" it always confused me (with the above "style"), it should have a separate name, like "marker" (or even a "mark" or "marked tekst"), and we will have `Jump Down >> Mark/Marker (or To mark/marker, or Find mark/marker)` and `Copy to Clipobard >> All (or Find) marks/markers`. But these are just my feelings.

Backing to picture, not bad, but above `to Clipboard` (for path) are necessary (will stay)?

**Comment by sasumner** [comment]
@ArkadiuszMichalski 

So the text in the English UI is what all of the translations are based upon.
Tweaking the established UI therefore gets things out of sync with the translations.
So therefore, we tend to leave the English UI alone (we try to make good choices at the start).

@donho says this is not an absolute rule (meaning that if the UI text is really bad. let's change it).

But it makes sense for the translation reason and for the user manual reasons as well.

So, yes, "Find style" is not the best text.
And, yes, "...to Clipboard" will stay for those that already have it.

I don't even like the phrase "marked text".
I would rather see "red-marked text" but since the color is configurable that doesn't make sense either.

BTW, in keeping with how _Search_ > _Unmark All_ > _Clear all Styles_ works, the new _Edit_ > _Copy to Clipboard_ > _All Styles text_ will **_NOT_** copy the (red)marked text.

**Comment by ArkadiuszMichalski** [comment]
> BTW, in keeping with how _Search_ > _Unmark All_ > _Clear all Styles_ works, the new _Edit_ > _Copy to Clipboard_ > _All Styles text_ will **_NOT_** copy the (red)marked text.

Thats why this two functionalities should have completely separate names (everywhere) so as not to be confusing. Maybe it's too late to change, maybe not. I change it anyway in the local translation, because it does not have a broader impact on a larger scale (like manual). 

BTW, in Mark window we have `Clear all marks` button, so `Copy to Clipboard > All marks` seems consistent. In other site in Search we have group Mark/Unmark all for Style, again inconsistent... and the circle keeps spinning. But I'm not complaining anymore, the name can always be changed, it's important to implement it first.

**Comment by sasumner** [comment]
>and the circle keeps spinning.

I definitely agree.
My attempt is to make new things (slightly) better, without going too far outside of any established convention.

Anyway, I do what I think is best at the time of implementation.
If Don likes it, he accepts, otherwise he asks for change.

**Comment by donho** [comment]
English is not my native language, so I don't talk about it for the moment.
For the place of commands, I would vote for adding these command into "Copy to Clipboard" folder and put this folder into the group of "Mark all", "Unmark all" in `Search` menu. For 2 reasons:
1. easy to discover and reuse these commands.
2. it's cleaner than putting them into `Edit >> Copy to Clipboard`.

**Comment by donho** [comment]
@ArkadiuszMichalski 
> BTW, in Mark window we have Clear all marks button, so Copy to Clipboard > All marks seems consistent.

That'll make 3 levels of depth - it's quite bad for user experience IMO

> In other site in Search we have group Mark/Unmark all for Style, again inconsistent... 

At least, users know where to find all the mark operations.

**Comment by ArkadiuszMichalski** [comment]
Hmm 3 levels depth? I just suggest `(Find) Marked test` in https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8943#issuecomment-702958710 change to `All marks` (in Mark window whe have button `Mark All` and `Clear all marks`, so jumping to `Copy to Clipboard > All marks` seems consistent, regardles where you put this `Copy to Clipboard`).

But this group Mark All/Unmark All fol Style 1-5 (in Search menu) have the same command like context menu (and here we have the names for this group `Style token` and `Remove style`). It's a bit inconsistent.

**Comment by donho** [comment]
@ArkadiuszMichalski OK, you're talking about naming. My bad.

**Comment by ArkadiuszMichalski** [comment]
Yes, I only add that these are two separate marking mechanisms (they are added/removed  differently) and word mixing style / mark (for command names and even for groups) is a bit confusing (especially at the beginning). The plus is that for Mark the command is always behind Style1-5 (so it's name is not so important).

Maybe to better differentiate between these two mechanisms it is worth adding a separator.
![test](https://user-images.githubusercontent.com/2730894/94989504-a0156000-0575-11eb-9441-19c567af21c3.gif)

and more correct is not `Found style`?

I mention about this because this bug put another commands, and it would be nice to use some consistent names. But as I wrote, it can even be improved later. That's why I don't comment on names anymore :) 

Backing to place, for me it makes no difference, but bookmarks (and copying them) are placed in Search, so this may also sit here (will be more consistent).

**Comment by sasumner** [comment]
@donho @ArkadiuszMichalski 

OK, so here is what I will do:

* Put these items on the _Search_ menu (near the related ones), per Don's request ( https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8943#issuecomment-703016811 )

* Provide the "best" English text that I can, with the "best" balance (my opinion) of matching existing terminology --but yet pushing towards better new terminology where possible.

After that there can be "arguments" about what UI text should go in the final version, and I will modify the UI text I had been using for final submission of a PR.

I know that Don is not a native English speaker.

@ArkadiuszMichalski I'm not positive, although it is my strong feeling that English is not your FIRST language.

English is my ONLY language (well, not counting C++/Python/etc), so when I put forth some UI text, you can believe that it is the best English translation possible -- within reason, but this is reasonable to assume when we are talking about text on a UI, not writing papers for college.  :-)

**Comment by ArkadiuszMichalski** [comment]
I'm not native, and if you think a given name is better then it probably is. But at first glance, when I saw `Find style`, I thought it means jump to 1st or 2nd or 3rd or 4rd or 5th style (interestingly, there is no such command, and maybe it should be), I didn't expect this to be the case for marks (added by Mark window).

**Comment by donho** [comment]
>  when I saw Find style, I thought it means jump to 1st or 2nd or 3rd or 4rd or 5th style 

Agree. I have had no idea about "Find style" due to lake of context: for us as coder, we know it's the same, by its nature, as the other styles so we put them together and name it "Find Style". But it's hard for users to have such association.

"Found & Marked style" might be more explicit, but the context is still missing, due to it's emplacement.

**Comment by sasumner** [comment]
@donho said:

>I have had no idea about "Find style" when I saw this term at the 1st time

OK, so _Find style_ has been that way since at least 6.7.5 (my first used version) and probably much earlier.
Are you saying YOU coded it but have had no idea about it? :-)
OR, someone else coded it and it confused you and yet you approved/merged their PR? :-)

To be honest, I discovered jumping to "Find style" when @guy038 pointed it out in a Community posting, after I had been using Notepad++ for a long time already.
Before that I had no idea what it meant.  
Of course I had tried running that command, but as I had no (red)marked text at the time, running the command did nothing.

So, in conclusion, bad UI text can hide features that users might really want to use, but cannot because they don't know the features exist.

But there still remains the task of creating new UI text and yet keeping some commonality with past text -- challenging. :-)
